app/scripts/lib/Stores/ExtensionStore.ts

```diff
@@ -0,0 +1,70 @@
+import browser from 'webextension-polyfill';
+import log from 'loglevel';
+import { checkForLastError } from '../../../../shared/modules/browser-runtime.utils';
+import {
+  type IntermediaryStateType,
+  MetaMaskStorageStructure,
+  BaseStore,
+} from './BaseStore';
+
+/**
+ * An implementation of the MetaMask Extension BaseStore system that uses the
+ * browser.storage.local API to persist and retrieve state.
+ */
+export default class ExtensionStore extends BaseStore {
+  isSupported: boolean;
+
+  constructor() {
+    super();
+    this.isSupported = Boolean(browser.storage.local);
+    if (!this.isSupported) {
+      log.error('Storage local API not available.');
+    }
+  }
+
+  /**
+   * Returns all of the keys currently saved
+   *
+   * @returns the key-value map from local storage
+   */
+  async get(): Promise<MetaMaskStorageStructure | null> {
+    if (!this.isSupported) {
+      log.error('Storage local API not available.');
+      return null;
+    }
+    const { local } = browser.storage;
+    const result = await local.get(null);
+    const err = checkForLastError();
```

Comment:
For the older callback-based extension APIs, the only way to check for an error was to check a global `lastError` variable. That's the pattern captured in this `checkForLastError` function.

I see that you've switched this to using the Promise-based API, so we don't need this anymore. `local.get` will resolve to a rejected Promise when there is an error.

Reply:
Simplified as per your comment in 46d664a000